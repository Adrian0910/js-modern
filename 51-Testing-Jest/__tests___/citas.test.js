import { expect } from '@jest/globals';
import Citas from '../js/classes/Citas'

describe('Probar la clase de citas', () => {
    
    const citas = new Citas();

    const id = Date.now();
    
    test('Agregar nueva cita', () => {
        const citaObj = {
            id,
            mascota: 'Hook',
            propietario: 'Adrian',
            telefono: '53269877',
            fecha: '10-12-2020',
            hora:'10:43',
            sintomas: 'soilo duerme'
        };
        citaObj.id = DAte.now();

        citas.agregarCita(citaObj);

        //Prueba
        expect(citas).toMatchSnapshot();
    });


    test('Actualizar cita', () => {
        const citaActualizada = {
            id,
            mascota: 'Nuevo nombre',
            propietario: 'Adrian',
            telefono: '53269877',
            fecha: '10-12-2020',
            hora:'10:43',
            sintomas: 'soilo duerme'
        };

        citas.editarCita(citaActualizada);

        expect(citas).toMatchSnapshot();
    });
    test('Eliminar cita', () => {
        citas.eliminarCita(id);

        expect(citas).toMatchSnapshot();
    });
});



