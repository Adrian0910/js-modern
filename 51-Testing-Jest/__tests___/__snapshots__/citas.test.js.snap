// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probar la clase de citas Actualizar cita 1`] = `
Citas {
  "citas": Array [],
}
`;

exports[`Probar la clase de citas Eliminar cita 1`] = `
Citas {
  "citas": Array [],
}
`;
